lib/UserBackend.php

```diff
@@ -148,10 +149,63 @@ public function createUserIfNotExists($uid, array $attributes = []) {
 			}
 			$qb->execute();
 
+			// If we use per-user encryption the keys must be initialized first
+			$userSecret = $this->getUserSecret($uid, $attributes);
+			if ($userSecret !== null) {
+				$this->updateUserSecretHash($uid, $userSecret);
+				// Emit a post login action to initialize the encryption module with the user secret provided by the idp.
+				\OC_Hook::emit('OC_User', 'post_login', ['run' => true, 'uid' => $uid, 'password' => $userSecret, 'isTokenLogin' => false]);
```

Comment:
please dispatch the typed event `\OCP\User\Events\PostLoginEvent` instead of deprecated `OC_Hook::emit(…)`.